controllers/istio_controller_test.go

```diff
@@ -50,6 +50,33 @@ var _ = Describe("IstioController", Ordered, func() {
 	cniObjectKey := client.ObjectKey{Name: "istio-cni-node", Namespace: istioNamespace}
 	webhookObjectKey := client.ObjectKey{Name: "istio-sidecar-injector-" + istioNamespace}
 
+	istioMeshConfigValues := string(`{
+		"accessLogFile": "/dev/stdout",
+    "enableTracing": true
+	}`)
+
+	istioPilotValues := string(`{
+		"image":"` + pilotImage + `",
+		"env": {
+			"PILOT_ENABLE_GATEWAY_API": true, 
+			"PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER": true,
+			"PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER": false,
+			"PILOT_ENABLE_GATEWAY_API_STATUS": true
+		}
+	}`)
+
+	istioCNIValues := string(`{
+		"enabled":true,
+		"excludeNamespaces": ["openshift", "kube-system"]
+	}`)
+
+	istioValues := []byte(`{
+		"donotexist": "shouldbeomitted",
```

Comment:
This value should throw an error instead of being silently ignored.

Reply:
Actually, this value is omitted without throwing an error by the controller-runtime client but only as an info warning `INFO    KubeAPIWarningLogger    unknown field "spec.values.donotexist"`. Should we try to catch this as an error?